sympy/holonomic/holonomic.py

```diff
@@ -592,7 +639,37 @@ def __add__(self, other):
                     else:
                         return self.change_ics(other.x0) + other
 
-        return HolonomicFunction(sol, self.x)
+        if self.x0 != other.x0:
+            return HolonomicFunction(sol, self.x)
+
+        # if the functions have singular_ics
+        y1 = None
+        y2 = None
+
+        if self.is_singularics() == False and other.is_singularics() == True:
```

Comment:
Correct. However `.is_singularics()` might return `None` if the function doesn't have initial condition, so using `if not self.is_singularics()` will become `true` in that case also.